tests/e2e/specs/backend/product-results-count.test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import {
+	canvas,
+	createNewPost,
+	insertBlock,
+	switchUserToAdmin,
+} from '@wordpress/e2e-test-utils';
+import { searchForBlock } from '@wordpress/e2e-test-utils/build/inserter';
+
+/**
+ * Internal dependencies
+ */
+import {
+	filterCurrentBlocks,
+	goToSiteEditor,
+	useTheme,
+	waitForCanvas,
+} from '../../utils.js';
+
+const block = {
+	name: 'Product Results Count',
+	slug: 'woocommerce/product-results-count',
+	class: '.wc-block-product-results-count',
+};
+
+describe( `${ block.name } Block`, () => {
+	describe( 'in a post', () => {
+		beforeAll( async () => {
+			await switchUserToAdmin();
+		} );
+
+		it( 'can not be inserted', async () => {
+			await createNewPost( {
+				postType: 'post',
+				title: block.name,
+			} );
+			await searchForBlock( block.name );
+			expect( page ).toMatch( 'No results found.' );
+		} );
+	} );
```

Comment:
Same as https://github.com/woocommerce/woocommerce-blocks/pull/8157#discussion_r1071140277, I think this can be simplified to one `it` block.